utils/vim/syntax/mlir.vim

```diff
@@ -31,7 +31,11 @@ syn match mlirType /x\s*\zsvector/
 " Operations.
 " Core ops (not exhaustive yet).
```

Comment:
I'd rather consider introducing some regexp that try to maintain a list of defined (std) ops. Something like

```
symbol ::= ^\s+[A-Za-z][A-Za-z0-9_]
dialect-prefix ::= ^\s+[A-Za-z][A-Za-z0-9_]*\.
dialect-symbol ::= dialect-prefix symbol
```

could work given that we tend to start new Ops from new lines.

Reply:
But we wouldn't want highlighting to work for typos like std.for etc. For now, I just included the more commonly used ones.